components/scream/src/physics/shoc/shoc_functions_f90.hpp

```diff
@@ -801,6 +850,122 @@ struct ShocMainData : public PhysicsTestData {
                     num_qtracers(num_qtracers_), dtime(dtime_), nbot_shoc(nbot_shoc_), ntop_shoc(ntop_shoc_) {}
 
   PTD_STD_DEF(ShocMainData, 8, shcol, nlev, nlevi, num_qtracers, dtime, nadv, nbot_shoc, ntop_shoc);
+
+  template <size_t N>
+  Real interpolate_data(const std::array<Real, N>& ref_elevations,
```

Comment:
This and compute_column_pressure are copied from shoc_ic_cases.cpp with minor changes to do all columns in a single call.